src/utilities/include/par.h

```diff
@@ -49,47 +53,251 @@ inline constexpr ExecutionPolicy autoPolicy(size_t size) {
   return ExecutionPolicy::Par;
 }
 
-#ifdef HAS_PAR_UNSEQ
-#define STL_DYNAMIC_BACKEND(NAME, RET)                        \
-  template <typename Ret = RET, typename... Args>             \
-  Ret NAME(ExecutionPolicy policy, Args... args) {            \
-    switch (policy) {                                         \
-      case ExecutionPolicy::Par:                              \
-        return std::NAME(std::execution::par_unseq, args...); \
-      case ExecutionPolicy::Seq:                              \
-        break;                                                \
-    }                                                         \
-    return std::NAME(args...);                                \
+template <typename Iter, typename F>
+void for_each(ExecutionPolicy policy, Iter first, Iter last, F f) {
+#if MANIFOLD_PAR == 'T'
+  if (policy == ExecutionPolicy::Par) {
+    tbb::parallel_for(tbb::blocked_range<Iter>(first, last),
+                      [&f](const tbb::blocked_range<Iter> &range) {
+                        for (Iter i = range.begin(); i != range.end(); i++)
+                          f(*i);
+                      });
+    return;
   }
-#define STL_DYNAMIC_BACKEND_VOID(NAME)                 \
-  template <typename... Args>                          \
-  void NAME(ExecutionPolicy policy, Args... args) {    \
-    switch (policy) {                                  \
-      case ExecutionPolicy::Par:                       \
-        std::NAME(std::execution::par_unseq, args...); \
-        break;                                         \
-      case ExecutionPolicy::Seq:                       \
-        std::NAME(args...);                            \
-        break;                                         \
-    }                                                  \
+#endif
+  std::for_each(first, last, f);
+}
+
+template <typename Iter, typename F>
+void for_each_n(ExecutionPolicy policy, Iter first, size_t n, F f) {
+  for_each(policy, first, first + n, f);
+}
+
+template <typename InputIter, typename BinaryOp,
+          typename T = typename std::iterator_traits<InputIter>::value_type>
+T reduce(ExecutionPolicy policy, InputIter first, InputIter last, T init,
+         BinaryOp f) {
+#if MANIFOLD_PAR == 'T'
+  if (policy == ExecutionPolicy::Par) {
+    // should we use deterministic reduce here?
```

Comment:
I would tend to say no, but I suppose it depends how much performance penalty there really is. It still won't be identical between different architectures for floating-point answers.

Reply:
Agreed, `deterministic` feels more like a compile option to me. Honestly I'd prefer to get rid of it entirely. Might be good to talk about use cases and what we're really trying to accomplish with it.

Reply:
We probably want to ask the openscad folks. I think I added that option when someone complained about non-deterministic results and I thought we can have deterministic results even with parallelization.

Making it a compile-time config will probably be a bit too limiting, e.g. openscad cannot have that as optional feature setting.

Reply:
@kintel Do you make use of this `deterministic` option? I don't love it, and I wonder if it's really necessary.

Reply:
Though for the record I think I *told* someone it can be used, I don't think openscad really added that option.